Resources/Private/JavaScript/Host/Containers/Drawer/index.js

```diff
@@ -15,23 +15,83 @@ import style from './style.css';
 const moduleLabel = (label, sourceName = 'Main') =>
     <I18n id={label} sourceName={sourceName} fallback={label}/>;
 
+const TARGET_WINDOW = 'Window';
+const TARGET_CONTENT_CANVAS = 'ContentCanvas';
+const THRESHOLD_MOUSE_LEAVE = 500;
+
 @connect($transform({
     isHidden: $get('ui.drawer.isHidden')
 }), {
-    hideDrawer: actions.UI.Drawer.hide
+    hideDrawer: actions.UI.Drawer.hide,
+    setContentCanvasSrc: actions.UI.ContentCanvas.setSrc
 })
 export default class Drawer extends Component {
     static propTypes = {
         isHidden: PropTypes.bool.isRequired,
-        hideDrawer: PropTypes.func.isRequired
+        hideDrawer: PropTypes.func.isRequired,
+        setContentCanvasSrc: PropTypes.func.isRequired,
+
+        menuData: PropTypes.objectOf(
+            PropTypes.shape({
+                icon: PropTypes.string,
+                label: PropTypes.string.isRequired,
+                uri: PropTypes.string.isRequired,
+                target: PropTypes.string,
+
+                children: PropTypes.arrayOf(
+                    PropTypes.shape({
+                        icon: PropTypes.string,
+                        label: PropTypes.string.isRequired,
+                        uri: PropTypes.string.isRequired,
+                        target: PropTypes.string,
+                        isActive: PropTypes.bool.isReqired
+                    })
+                )
+            })
+        ).isRequired
+    };
+
+    state = {
+        mouseLeaveTimeout: null
     };
 
-    shouldComponentUpdate(...args) {
-        return shallowCompare(this, ...args);
+    shouldComponentUpdate(nextProps) {
+        return shallowCompare(this.props, nextProps);
+    }
+
+    hideDrawer() {
+        const {hideDrawer} = this.props;
+        const {mouseLeaveTimeout} = this.state;
+
+        if (!mouseLeaveTimeout) {
+            const timeout = setTimeout(() => {
+                hideDrawer();
+                this.setState({
+                    mouseLeaveTimeout: null
+                });
+            }, THRESHOLD_MOUSE_LEAVE);
+
+            this.setState({
+                mouseLeaveTimeout: timeout
+            });
+        }
+    }
+
+    clearMouseLeaveTimeout() {
+        const {mouseLeaveTimeout} = this.state;
+
+        if (mouseLeaveTimeout) {
+            clearTimeout(mouseLeaveTimeout);
+            this.setState({
+                mouseLeaveTimeout: null
+            });
+        }
     }
 
     render() {
-        const {isHidden, hideDrawer} = this.props;
+        const {isHidden} = this.props;
+        const clearMouseLeaveTimeout = ::this.clearMouseLeaveTimeout;
```

Comment:
You're absolutely right. I have never thought about that issue tbh, but makes a lot of sense. Probably it would be best to move those bindings to the constructor then. wdyt?